tqp/topic_queue_poller.py

```diff
@@ -17,27 +17,43 @@ def _jsonify_dictionary(dictionary):
 # -----------------------------------------------------------------------------
 
 
-def create_queue(queue_name, *, tags, **kwargs):
-    sqs = boto3.resource("sqs")
+def _create_queue_raw(name, attributes, *, tags):
+    sqs_client = boto3.client("sqs")
+    sqs_resource = boto3.resource("sqs")
 
-    def _create_queue(name, attributes, extra_tags=None):
-        extra_tags = extra_tags or {}
+    attributes = _jsonify_dictionary(attributes)
+    tags = {"tqp": "true", **tags}
 
-        return sqs.create_queue(
-            QueueName=name,
-            Attributes=_jsonify_dictionary(attributes),
-            tags={"tqp": "true", "dlq": "false", **tags, **extra_tags},
+    def _create_queue():
+        return sqs_resource.create_queue(
+            QueueName=name, Attributes=attributes, tags=tags,
         )
 
-    dead_letter_queue = _create_queue(
+    try:
+        return _create_queue()
+    except sqs_client.exceptions.QueueNameExists:
+        queue_url = sqs_resource.get_queue_by_name(QueueName=name).url
+        existing_tags = sqs_client.list_queue_tags(QueueUrl=queue_url)["Tags"]
+        tags_to_remove = list(set(existing_tags.keys()) - set(tags.keys()))
+
+        sqs_client.tag_queue(QueueUrl=queue_url, Tags=tags)
+        if tags_to_remove:
+            sqs_client.untag_queue(QueueUrl=queue_url, TagKeys=tags_to_remove)
```

Comment:
might we not also need to update values on existing tags?

Reply:
`tag_queue` at line 39 does that